nova/conf/base.py

```diff
@@ -23,6 +23,19 @@
         default=12,
         min=0,
         help='Length of generated instance admin passwords.'),
+    cfg.IntOpt(
+        'password_all_group_samples',
+        default=1,
+        min=0,
+        help='''
+How often should the symbols be sampled from all groups
+to ensure the presence of all of them
+* Zero means purely random, so least predictable,
+  but possibly non-confirming to some password policies
+* One means, there will be at least one lower-case, one upper-case
```

Comment:
"any positive number" -> "at least that many"

How does that interact with the `password_length` setting? Should we document here, that the password might get longer or should we raise an error, if the settings do not fit with each other (e.g. password_length of 7 but password_all_group_samples of 3)?

Reply:
If  `password_length < password_all_group_samples * len(symbolgroups))` then the password will be cutted to `password_length` characters, thereby possibly reducing the number of characters out of the classes.

I'll add that to the documentation

I would not raise an error, as that would change the behaviour . Currently you also could set the password-length to two, (even zero for some reason), and it would just work.